runtime/vm/JFRWriter.hpp

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2024
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+#if !defined(JFRWRITER_HPP_)
+#define JFRWRITER_HPP_
+
+#include <stdlib.h>
+
+#include "j9cfg.h"
+#include "j9.h"
+#include "vm_api.h"
+
+extern "C" {
+class VM_JFRWriter {
+	/*
+	 * Data members
+	 */
+private:
+
+protected:
+
+public:
+	static constexpr const char* METADATA_BLOB_FILE_EMVVAR = "OPENJ9_METADATA_BLOB_FILE_PATH";
+
+	static constexpr const char* DEFAULT_JFR_FILE_NAME = "defaultJ9recording.jfr";
+
+
+	/*
+	 * Function members
+	 */
+private:
+	static void
+	closeJFRFile(J9JavaVM *vm)
+	{
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		if (-1 != vm->jfrState.blobFileDescriptor) {
+			j9file_close(vm->jfrState.blobFileDescriptor);
+		}
+	}
+
+	static bool
+	loadJFRMetadataBlob(J9JavaVM *vm)
```

Comment:
What is this file?

Reply:
its a metadata file that describes the format of the JFR events. Its 3) Metadata in https://www.morling.dev/blog/jdk-flight-recorder-file-format/.

Long term we will generate this as part of the build process

Reply:
We will need a way to handle cases where it doesn't exist. It will take some time before we have a solution that is integrated in the build system.

Reply:
Until then, let's not break JFR startup.

Reply:
Agreed. I've modified it such that startup will not be affected.